docs/guides/tutorials/nextjs_app_router.rst

```diff
@@ -0,0 +1,627 @@
+  ....... };
+  {default::BlogPost {id: 7f301d02-c780-11ec-8a1a-a34776e884a0}}
+  edgedb> insert BlogPost {
+  .......   title := "How to build a blog with EdgeDB and Next.js",
+  .......   content := "Let's start by scaffolding our app..."
+  ....... };
+  {default::BlogPost {id: 88c800e6-c780-11ec-8a1a-b3a3020189dd}}
+
+
+Loading posts with React Server Components
+------------------------------------------
+
+Now that we have a couple posts in the database, let's load them into our 
+Next.js app.
+To do that, we'll need the ``edgedb`` client library. Let's install that from
+NPM:
+
+.. code-block:: bash
+
+  $ npm install edgedb 
+  # or yarn add edgedb or pnpm add edgedb or bun add edgedb
+
+Then go to the ``app/page.tsx`` file to replace the static data with a
+the blogposts fetched from the database.
+
+To fetch these from the homepage, we'll create an EdgeDB client and use the
+``.query()`` method to fetch all the posts in the database with a 
+``select`` statement.
+
+.. code-block:: tsx-diff
+
+    // app/page.tsx
+
+    import Link from 'next/link'
+  + import { createClient } from 'edgedb';
+
+    type Post = {
+      id: string
+      title: string
+      content: string
+    }
+  + const client = createClient();
+
+    export default async function Home() {
+  -   const posts: Post[] = [
+  -     {
+  -       id: 'post1',
+  -       title: 'This one weird trick makes using databases fun',
+  -       content: 'Use EdgeDB',
+  -     },
+  -     {
+  -       id: 'post2',
+  -       title: 'How to build a blog with EdgeDB and Next.js',
+  -       content: "Start by scaffolding our app with `create-next-app`.",
+  -     },
+  -   ]
+  +   const posts = await client.query<Post>(`\
+  +    select BlogPost {
+  +      id,
+  +      title,
+  +      content
+  +   };`)
+
+      return (
+        <div className="container mx-auto p-4 bg-black text-white">
+          <h1 className="text-3xl font-bold mb-4">Posts</h1>
+          <ul>
+            {posts.map((post) => (
+              <li
+                key={post.id}
+                className="mb-4"
+              >
+                <Link
+                  href={`/post/${post.id}`}
+                  className="text-blue-500"
+                >
+                  {post.title}
+                </Link>
+              </li>
+            ))}
+          </ul>
+        </div>
+      )
+    }
+
+When you refresh the page, you should see the blog posts.
+
+Generating the query builder
+----------------------------
+
+Since we're using TypeScript, it makes sense to use EdgeDB's powerful query
+builder. This provides a schema-aware client API that makes writing strongly
+typed EdgeQL queries easy and painless. The result type of our queries will be
+automatically inferred, so we won't need to manually type something like
+``type Post = { id: string; ... }``.
+
+First, install the generator to your project.
+
+.. code-block:: bash
+
+  $ yarn add --dev @edgedb/generate
```

Comment:
We should probably stick with one package manager/script runner as the recommended choice. We start by recommending `yarn dev` to start the dev server before switching to `npm install edgedb` and then back to `yarn add` here. I would lean toward `npm` as the default recommendation since users who meet the Node prerequisite will already have that installed as well, but I'm fine with `yarn` if there's a compelling case for it in the context of this tutorial.

Reply:
Hmm, it actually only matters for the install commands. I used a different style here as the command was longer than the others.

Reply:
Maybe if they have both Yarn and npm, it doesn't matter which they use to install a dependency, but it _will_ matter if they *don't* have Yarn installed. Should we assume they do? If so, should we maybe list pre-requisites at the top of the guide, including Yarn among them? Should we take them through the steps to install it or maybe link them to the Yarn installation instructions?

I suggest we avoid all those questions by showing everything with npm by default which we _know_ they have if they have Node.

Reply:
> I suggest we avoid all those questions by showing everything with npm by default which we know they have if they have Node.

I think this is a good fallback solution. Will go with this one.